tests/test_integration_upload.py

```diff
@@ -13,6 +13,23 @@
 client = skynet.SkynetClient()
 
 
+def response_callback(request):
```

Comment:
I don't understand what this function does or why we started using it as a callback. Can you please add a docstring for the function and some comments?

Reply:
it's a way responses has for handling requests.  it could be used to do server actions like providing the appropriate skylink and response for the data.  i use it to simply gather the uploaded data so it can be verified the code under test uploaded the right data.